src/data/validate.rs

```diff
@@ -0,0 +1,365 @@
+use crate::config::Mapping;
+
+/// Validate paths for entity
+pub fn validate_paths_for_entity(
+    entity: &str,
+    mappings: &Vec<Mapping>,
+    api_schema: &serde_json::Map<String, serde_json::Value>,
+) -> anyhow::Result<()> {
+    // if entity name is not set in api_schema throw an exception
+    if !api_schema.contains_key(entity) {
+        anyhow::bail!("Entity {} not found in API schema", entity);
+    }
+
+    for entry in mappings {
+        let path_mapping = match entry {
+            Mapping::ByPath(path_mapping) => path_mapping,
+            Mapping::ByScript(_) => continue,
+        };
+
+        let path = path_mapping.entity_path.split('.').collect::<Vec<_>>();
+
+        // if path starts with ? its optional
+        if path[0].ends_with('?') {
+            continue;
+        }
+
+        // check if first path element is set in the entity object in schema
+        if !api_schema[entity]["properties"][path[0]].is_object() {
+            anyhow::bail!("Entity {} does not have a field {}", entity, path[0]);
+        }
+
+        let root_path = api_schema[entity]["properties"][path[0]].as_object().unwrap();
+
+        // if there is only one path element
+        if path.len() == 1 {
+            let field_type = root_path["type"].as_str().unwrap();
```

Comment:
Some notes on error handling in Rust, you can either:

## Panic

which kills the whole application with a message and should generally only be done if you can't recover from an error.

## Return an error with the `Result` type or an optional value with the `Option` type

As this is very common in Rust, there is even a nice shorthand: the question mark operator. See for more information
https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator

In this case `as_str()`returns an `Option<&str>`, but your function returns a `Result<(), anyhow::Error>`, so just using `.as_str()?` doesn't work, because rust doesn't know which error to return. But you convert the `Option` into an `Result`, for example like so:
```rust
let field_type = root_path["type"].as_str().context("'type' isn't a string")?; // utility trait provided by `anyhow`, might need an import here
```


## There is no try / catch / exceptions

So every function that can fail, can easily be identified as such by it's return type. If a function can panic, it's good practice to write it inside the doc comment of that function under which circumstances it panics.